packages/react/src/lib/positioner/Popper.tsx

```diff
@@ -0,0 +1,171 @@
+   */
+  position?: Position
+
+  /**
+   * Array of conditions to be met in order to trigger a subsequent render to reposition the elements.
+   */
+  positioningDependencies?: any[]
+}
+
+export interface PopperChildrenProps {
+  /**
+   * Popper's placement.
+   */
+  placement: PopperJS.Placement
+}
+
+type PopperChildrenFn = (props: PopperChildrenProps) => React.ReactNode
+
+interface PopperProps extends PositionCommonProps {
+  /**
+   * Ref object containing the pointer node.
+   */
+  pointerRef?: React.RefObject<Element>
+
+  /**
+   * The content of the Popper box (the element that is going to be repositioned).
+   */
+  children: PopperChildrenFn | React.ReactNode
+
+  /**
+   * Enables events (resize, scroll).
+   * @prop {Boolean} eventsEnabled=true
+   */
+  eventsEnabled?: boolean
+
+  /**
+   * List of modifiers used to modify the offsets before they are applied to the Popper box.
+   * They provide most of the functionality of Popper.js.
+   */
+  modifiers?: PopperJS.Modifiers
+
+  /**
+   * Enables the Popper box to position itself in 'fixed' mode (default value is position: 'absolute')
+   * @prop {Boolean} positionFixed=false
+   */
+  positionFixed?: boolean
+
+  /**
+   * Ref object containing the target node (the element that we're using as reference for Popper box).
+   */
+  targetRef?: React.RefObject<Element>
+
+  /**
+   * Rtl attribute for the component.
+   */
+  rtl?: boolean
+
+  /**
+   * Array of conditions to be met in order to trigger a subsequent render to reposition the elements.
+   */
+  positioningDependencies?: any[]
+}
+
+/**
+ * Popper relies on the 3rd party library [Popper.js](https://github.com/FezVrasta/popper.js) for positioning.
+ */
+const Popper: React.FunctionComponent<PopperProps> = props => {
+  const {
+    align,
+    children,
+    eventsEnabled,
+    modifiers,
+    offset,
+    pointerRef,
+    position,
+    positionFixed,
+    positioningDependencies,
+    rtl,
+    targetRef,
+  } = props
+
+  const proposedPlacement = getPlacement({ align, position, rtl })
+
+  const popperRef = React.useRef<PopperJS>()
+  const contentRef = React.useRef<HTMLElement>(null)
+  const latestPlacement = React.useRef<PopperJS.Placement>(proposedPlacement)
+  const [computedPlacement, setComputedPlacement] = React.useState<PopperJS.Placement>(
+    proposedPlacement,
+  )
+
+  React.useEffect(
+    () => {
+      const handleUpdate = (data: PopperJS.Data) => {
+        if (data.placement !== latestPlacement.current) {
+          latestPlacement.current = data.placement
+          setComputedPlacement(data.placement)
+        }
+      }
+
+      const pointerRefElement = pointerRef && pointerRef.current
+      const options: PopperJS.PopperOptions = {
+        placement: proposedPlacement,
+        eventsEnabled,
+        positionFixed,
+        modifiers: {
+          ...(offset && {
+            offset: { offset: rtl ? applyRtlToOffset(offset, position) : offset },
```

Comment:
I suggest to move out this to a root of function and call it `offsetModifier`, it will allow us to reduce dependency list

```ts
const offsetModifier = React.useMemo(() => {
  return { offset: rtl ? applyRtlToOffset(offset, position) : offset }
}, [rtl, offset, position])
```

Reply:
done a bit differently, thanks